sklearn/ensemble/gradient_boosting.py

```diff
@@ -956,8 +957,12 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # init state
             self._init_state()
 
-            # fit initial model - FIXME make sample_weight optional
-            self.init_.fit(X, y, sample_weight)
+            # fit initial model
+            # check if init accepts sample_weight argument
+            if 'sample_weight' in getargspec(self.init_.fit).args:
```

Comment:
We have an utility function for that
https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/utils/validation.py#L548